lxd/auth/oidc/oidc.go

```diff
@@ -371,14 +371,36 @@ func (*Verifier) setCookies(w http.ResponseWriter, idToken string, refreshToken
 	return nil
 }
 
+// Opts contains optional configurable fields for the Verifier. A variadic of func(*Opts) can
+// be passed into NewVerifier to apply these configuration options.
+type Opts struct {
+	configExpiryInterval time.Duration
+}
+
+// WithConfigExpiryInterval can be passed into LoadAuthorizer to pass in driver specific configuration.
+func WithConfigExpiryInterval(interval time.Duration) func(*Opts) {
+	return func(o *Opts) {
+		o.configExpiryInterval = interval
+	}
+}
+
 // NewVerifier returns a Verifier.
-func NewVerifier(issuer string, clientid string, audience string) (*Verifier, error) {
-	cookieKey, err := uuid.New().MarshalBinary()
-	if err != nil {
-		return nil, fmt.Errorf("Failed to create UUID: %w", err)
+func NewVerifier(issuer string, clientID string, audience string, clusterCert *shared.CertInfo, options ...func(*Opts)) (*Verifier, error) {
```

Comment:
Can we pass the opts directly as an arg rather than a variadic so we can easily see the available options in the function definition?

Reply:
Done.